crates/aptos-crypto/src/traits.rs

```diff
@@ -62,22 +62,31 @@ pub trait ValidCryptoMaterial:
     // The for<'a> exactly matches the assumption "deserializable from any lifetime".
     for<'a> TryFrom<&'a [u8], Error=CryptoMaterialError> + Serialize + DeserializeOwned
 {
+    /// Prefix for AIP-80 e.g. ed25519-priv
+    const AIP_80_PREFIX: &'static str;
+
     /// Convert the valid crypto material to bytes.
     fn to_bytes(&self) -> Vec<u8>;
 }
 
-/// An extension to to/from Strings for [`ValidCryptoMaterial`].
+/// An extension to/from Strings for [`ValidCryptoMaterial`].
 ///
 /// Relies on [`hex`] for string encoding / decoding.
 /// No required fields, provides a default implementation.
 pub trait ValidCryptoMaterialStringExt: ValidCryptoMaterial {
     /// When trying to convert from bytes, we simply decode the string into
     /// bytes before checking if we can convert.
     fn from_encoded_string(encoded_str: &str) -> std::result::Result<Self, CryptoMaterialError> {
+        let mut str = encoded_str;
+        // First strip the prefix
+        if Self::AIP_80_PREFIX.is_empty() && str.starts_with(Self::AIP_80_PREFIX) {
```

Comment:
Was this supposed to be checking: `if AIP-80 prefix is **not** empty`?

Reply:
As is, if the struct being deserialized has a (non-empty) AIP-80 prefix, it will **not** be stripped because the `is_empty` check above returns `false`, no?

Reply:
Yes, it's backwards.  Should be not empty.  Let me check this